ignite/metrics/gan/fid.py

```diff
@@ -0,0 +1,207 @@
+            Otherwise, default value is 2048.
+        feature_extractor: a callable for extracting the features from the input data. If neither num_features nor
+            feature_extractor are defined, default value is ``InceptionExtractor``.
+        output_transform: a callable that is used to transform the
+            :class:`~ignite.engine.engine.Engine`'s ``process_function``'s output into the
+            form expected by the metric. This can be useful if, for example, you have a multi-output model and
+            you want to compute the metric with respect to one of the outputs.
+            By default, metrics require the output as ``(y_pred, y)`` or ``{'y_pred': y_pred, 'y': y}``.
+        device: specifies which device updates are accumulated on. Setting the
+            metric's device to be the same as your ``update`` arguments ensures the ``update`` method is
+            non-blocking. By default, CPU.
+
+    Example:
+
+        .. code-block:: python
+
+            from ignite.metric.gan import FID
+            import torch
+            y_pred, y = torch.rand(10,2048), torch.rand(10,2048)
+            m = FID()
+            m.update((y_pred,y))
+            print(m.compute())
+
+    .. versionadded:: 0.5.0
+    """
+
+    def __init__(
+        self,
+        num_features: Optional[int] = None,
+        feature_extractor: Optional[Callable] = None,
+        output_transform: Callable = lambda x: x,
+        device: Union[str, torch.device] = torch.device("cpu"),
+    ) -> None:
+
+        try:
+            import scipy  # noqa: F401
+        except ImportError:
+            raise RuntimeError("This module requires scipy to be installed.")
+
+        # default is inception
+        if num_features is None and feature_extractor is None:
+            num_features = 2048
+            feature_extractor = InceptionExtractor()
+        elif num_features is None:
+            raise ValueError("num of features should be defined")
+        elif feature_extractor is None:
+            self._feature_extractor = lambda x: x
+            feature_extractor = self._feature_extractor
+
+        if num_features <= 0:
+            raise ValueError(f"num of features must be greater to zero (got: {num_features})")
+        self._num_features = num_features
+        self._feature_extractor = feature_extractor
+        self._eps = 1e-6
+        super(FID, self).__init__(output_transform=output_transform, device=device)
+
+    @staticmethod
+    def _online_update(features: torch.Tensor, total: torch.Tensor, sigma: torch.Tensor) -> None:
+        total += features
+        sigma += torch.outer(features, features)
+
+    def get_covariance(self, sigma: torch.Tensor, total: torch.Tensor) -> torch.Tensor:
+        r"""
+        Calculates covariance from mean and sum of products of variables
+        """
+        sub_matrix = torch.outer(total, total)
```

Comment:
`torch.outer` is not available for torch version < v1.7.0, we have to replace it with something else that supports torch v1.3.1 as min version.

Reply:
Before this it was refereed to as `torch.ger` in previous versions. But that will soon be depreciated.

Reply:
If we can't use a vectorized function to do it, we will have to use a for loop.
```
for i,feat in enumerate(features):
    sigma[i] += feat*features
```

Reply:
Use torch.ger for <1.7.0 and outer for others.

Reply:
Okay! I will change it!